src/test/java/tests/sandbox/SqlStorageIndexedTests.java

```diff
@@ -0,0 +1,196 @@
+package tests.sandbox;
+
+import junit.framework.TestCase;
+import org.commcare.cases.ledger.Ledger;
+import org.commcare.cases.model.Case;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import sandbox.JdbcSqlStorageIterator;
+import sandbox.SqlSandboxUtils;
+import sandbox.SqliteIndexedStorageUtility;
+import sandbox.UserSqlSandbox;
+
+import javax.sql.DataSource;
+import java.util.Vector;
+
+import static junit.framework.Assert.fail;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SqlStorageIndexedTests {
+
+    private Case a, b, c;
+
+    private Ledger l, l2, l3;
+
+    private SqliteIndexedStorageUtility<Case> caseStorage;
+    private SqliteIndexedStorageUtility<Ledger> ledgerStorage;
+    private String owner;
+    private String groupOwner;
+    private String otherOwner;
+    private Vector<String> groupOwned;
+    private Vector<String> userOwned;
+
+
+    @Before
+    public void setUp() throws Exception {
+
+        owner = "owner";
+        otherOwner = "otherowner";
+        groupOwner = "groupowned";
+
+        userOwned = new Vector<>();
+        userOwned.addElement(owner);
+
+        groupOwned = new Vector<>();
+        groupOwned.addElement(owner);
+        groupOwned.addElement(groupOwner);
+
+        a = new Case("a_case_name", "case_type_ipsum");
+        a.setCaseId("a_case_id");
+        a.setUserId(owner);
+        a.setID(-1);
+
+        b = new Case("b_case_name", "case_type_ipsum");
+        b.setCaseId("b_case_id");
+        b.setUserId(owner);
+        b.setID(-1);
+
+        c = new Case("c_case_name", "case_type_ipsum");
+        c.setCaseId("c_case_id");
+        c.setUserId(owner);
+        c.setID(-1);
+
+        l = new Ledger("ledger_entity_id");
+        l.setID(-1);
+        l.setEntry("test_section_id", "test_entry_id", 2345);
+
+        l2 = new Ledger("ledger_entity_id");
+        l2.setID(-1);
+        l2.setEntry("test_section_id_2", "test_entry_id_2", 2345);
+
+        l3 = new Ledger("ledger_entity_id_3");
+        l3.setID(-1);
+        l3.setEntry("test_section_id_3", "test_entry_id_3", 2345);
+    }
+
+    @Test
+    public void testSqlCaseStorage() {
```

Comment:
this test is pretty long and hard to understand what's being tested. maybe some comments?